make an option for dark bkg #2830

kpaharelau opened the pull request:
Use menu Notebook-Theme-Ambiance. I think that some issues will be found. I ready to fix its

kpaharelau committed:
make an option for dark bkg #2830

kpaharelau committed:
make an option for dark bkg #2830

kpaharelau committed:
make an option for dark bkg #2830

FIXED

spot [3:57 PM]
ok well 1 first is ok but do it in a way that just uses the css by index

cit [4:16 PM]
May be "by name" will be better solution?

spot [4:28 PM]
yes

scottdraves commented on core/src/main/web/app/beaker.js:
```diff
@@ -383,6 +383,7 @@
       $document.bind('drop', function() {
         $('body').removeClass('dragover');
       });
+      window.bkCoreManager = bkCoreManager;
```
what are these for?  looks bad...

scottdraves commented on core/src/main/web/app/styles/theme/ambianse.scss:
```diff
@@ -0,0 +1,256 @@
+$main-theme-panel-background: #202020;
```
needs copyright banner.  typo in file name.

tomca32 commented on core/src/main/web/app/helpers/core.js:
```diff
@@ -1033,6 +1053,27 @@
+        var classes = [];
+        if(ca && ca.length && ca.split){
+          ca = jQuery.trim(ca); /* strip leading and trailing spaces */
+          ca = ca.replace(/\s+/g,' '); /* remove doube spaces */
```
This is just a string of class names. Why do we expect spaces there?

tomca32 commented on core/src/main/web/app/helpers/core.js:
```diff
@@ -1033,6 +1053,27 @@
+        var classCount = classes.length;
+        for (var i = 0; i < classCount; i++) {
+          var clazz = classes[i];
+          if (clazz.indexOf("beaker-s") >= 0){
```
why "beaker-s"? Can we name it something like "beaker-theme" . It would also be good to put that string in globals.js and then inject here.

tomca32 commented on core/src/main/web/app/helpers/core.js:
```diff
@@ -1033,6 +1053,27 @@
+        }
+
+        var classCount = classes.length;
+        for (var i = 0; i < classCount; i++) {
```
We are basically doing a find here. This can be simplified a lot with _.find:

```
_.find(classes, function(e) {return e.indexOf(BEAKER_THEME) !== -1};)
```

tomca32 commented on core/src/main/web/app/helpers/core.js:
```diff
@@ -1033,6 +1053,27 @@
         });
+      },
+      colorize: function (theme) {
+        var colorizedElements = $("body, html");
```
why both body and html elements? Isn't just one of them enough?

tomca32 replied:
Also I feel like this whole function could be a directive on the body. What do you think?

kpaharelau committed:
make an option for dark bkg #2830

needs copyright banner. typo in file name. FIXED

@scottdravesscottdraves commented on the diff 15 hours ago
View full changes core/src/main/web/app/beaker.js
 @@ -383,6 +383,7 @@
       $document.bind('drop', function() {
         $('body').removeClass('dragover');
       });
+      window.bkCoreManager = bkCoreManager;
@scottdraves   Owner
scottdraves added a note 15 hours ago
what are these for? looks bad... FIXED

We are basically doing a find here. This can be simplified a lot with _.find:

_.find(classes, function(e) {return e.indexOf(BEAKER_THEME) !== -1};) Good idea :) - CHANGED

>>why both body and html elements? Isn't just one of them enough? - I tested that html is enough. CHANGED

why "beaker-s"? Can we name it something like "beaker-theme" . It would also be good to put that string in globals.js and then inject here. beaker-s(tyle) :-)

>>Also I feel like this whole function could be a directive on the body. What do you think?
>>It would also be good to put that string in globals.js and then inject here.

I think that this will complicate the code.

>>This is just a string of class names. Why do we expect spaces there?

I see this functionality in some examples from internet. I would have left this code.

kpaharelau committed:
make an option for dark bkg #2830

typo in file name. FIXED

tomca32 commented on core/src/main/web/app/helpers/core.js:
```diff
@@ -1033,6 +1053,22 @@
+      colorize: function (theme) {
+        var colorizedElements = $("html");
+        var ca = colorizedElements.attr('class');
+        var classes = [];
```
I think we don't need this line anymore since we can just initialize it when we do the split

tomca32 commented on core/src/main/web/app/helpers/core.js:
```diff
@@ -1033,6 +1053,22 @@
+        var classes = [];
+        if(ca && ca.length && ca.split){
+          ca = jQuery.trim(ca); /* strip leading and trailing spaces */
+          ca = ca.replace(/\s+/g,' '); /* remove double spaces */
```
If the trimming and removing double spaces is required I think it needs a comment explaining why since it's pretty surprising.

tomca32 commented on core/src/main/web/app/helpers/core.js:
```diff
@@ -1033,6 +1053,22 @@
+        }
+        var clazz = _.find(classes, function(e) {
+          return e.indexOf("beaker-s") !== -1
+        });
```
This looks much better.

kpaharelau committed:
make an option for dark bkg #2830

spot [5:41 PM]
ok def use "beaker-theme". FIXED

ok good just delete the space handling then FIXED

kpaharelau commented on core/src/main/web/app/helpers/core.js:
```diff
@@ -1033,6 +1053,24 @@
+          classes = ca.split(' ');
+        }
+        var themeStylePrefix = "beaker-theme-";
+        var clazz = _.find(classes, function (e) {
```
We can try to use undefined 'classes' in this place if remove 'var classes = []'.

tomca32 replied:
good point

Commit:
#2830 fixing ambiance theme styles

Commit:
#2830 fixing ambiance theme styles - selection fixes

Commit:
#2830 fixing ambiance theme styles

Commit:
#2830 fixing ambiance theme styles - CM fullscreen styles fix

Commit:
#2830 fixing ambiance theme styles

Commit:
#2830 fixing ambiance theme styles - fixing default buttons style

Commit:
#2830 fixing ambiance theme styles - fixing stdout/err styles

Commit:
#2830 fixing ambiance theme styles - fixing stdout/err styles

Commit:
#2830 fixing ambiance theme styles - fixing stdout/err styles
